File: tests/test-minecraft-osp/src/main/java/pize/tests/voxelgame/main/command/argument/CommandArgTime.java
package pize.tests.voxelgame.main.command.argument;

import pize.tests.voxelgame.main.time.TimeUnit;
import pize.tests.voxelgame.server.Server;
import pize.tests.voxelgame.main.command.source.CommandSource;

public class CommandArgTime extends CommandArg{
    
    // Результат парсинга
    private TimeUnit unit;
    private float time;
    
    @Override
    public int parse(String remainingChars, CommandSource source, Server server){
        // Разделяем оставшуюся часть команды на части
        final String[] args = remainingChars.split(" ");
        
        // Если количество частей меньше 1 (время), завершить парсинг
        if(args.length < 1)
            return 0;
        
        String timeString = args[0];
        
        // Устанавливаем еденицу измерения
        unit = null;
        if(timeString.endsWith("d")) unit = TimeUnit.DAYS;
        if(timeString.endsWith("m")) unit = TimeUnit.MINUTES;
        if(timeString.endsWith("s")) unit = TimeUnit.SECONDS;
        if(timeString.endsWith("t")) unit = TimeUnit.TICKS;

        if(unit != null)
            timeString = timeString.substring(0, timeString.length() - 1);
        else
            unit = TimeUnit.TICKS; // Ticks - default
        
        // Устанавливаем время
        try{
            time = Float.parseFloat(timeString);
        }catch(Exception ignored){
            return 0;
        }
        return args[0].length();
    }
    
    public TimeUnit getUnit(){
        return unit;
    }
    
    public float getTime(){
        return time;
    }
    
}

File: tests/test-minecraft-osp/src/main/kotlin/pize/tests/voxelgame/main/command/argument/CommandArgTime.kt
package pize.tests.voxelgame.main.command.argument

import pize.tests.voxelgame.main.command.source.CommandSource
import pize.tests.voxelgame.main.time.TimeUnit
import pize.tests.voxelgame.server.Server

class CommandArgTime : CommandArg() {
    // Результат парсинга
    var unit: TimeUnit? = null
        private set
    var time = 0f
        private set

    override fun parse(remainingChars: String, source: CommandSource, server: Server): Int {
        // Разделяем оставшуюся часть команды на части
        val args = remainingChars.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()

        // Если количество частей меньше 1 (время), завершить парсинг
        if (args.size < 1) return 0
        var timeString = args[0]

        // Устанавливаем еденицу измерения
        unit = null
        if (timeString.endsWith("d")) unit = TimeUnit.DAYS
        if (timeString.endsWith("m")) unit = TimeUnit.MINUTES
        if (timeString.endsWith("s")) unit = TimeUnit.SECONDS
        if (timeString.endsWith("t")) unit = TimeUnit.TICKS
        if (unit != null) timeString = timeString.substring(0, timeString.length - 1) else unit =
            TimeUnit.TICKS // Ticks - default

        // Устанавливаем время
        time = try {
            timeString.toFloat()
        } catch (ignored: Exception) {
            return 0
        }
        return args[0].length
    }
}
